### ISSUE TITLE
Edison module load error after maintainence on 7/25/2018

### ISSUE THREAD

**Issue by tangq** [E3SM-Project/E3SM#2463]
For the DECK runs with branch "maint-1.0" (v1.0.0-10-gdd18fc56e), I got the following error when submitting jobs. This prevents additional DECK runs on Edison. Tagging @golaz 

@ndkeen , could you check if it's possible to have the "maint-1.0" branch use the old modules so it BFB reproduces the DECK results? Thanks!

2018-07-25 20:57:01: case.submit error
ERROR: module command /opt/modules/default/bin/modulecmd python load cray-mpich/7.6.0 PrgEnv-intel/6.0.4 failed with message:
cray-mpich(3):ERROR:105: Unable to locate a modulefile for 'cray-mpich/7.6.0'

**Comment by ndkeen** [comment]
It looks like I need to adjust modules after update. I can also try to get the old modules to work. Currently on vacation, but maybe can do sooner

**Comment by tangq** [comment]
Thanks @ndkeen , I'd prefer to have the old modules to work since we may need more DECK runs in the future. It is important that those runs are done with the same modules.

I will hold on submitting new jobs till this module issue is solved.

**Comment by tangq** [comment]
Similar error fails the short-term archive:

ERROR: module command /opt/modules/default/bin/modulecmd python load cray-mpich/7.6.0 PrgEnv-intel/6.0.4 failed with message:
cray-mpich(3):ERROR:105: Unable to locate a modulefile for 'cray-mpich/7.6.0'

**Comment by darincomeau** [comment]
I have also just run into this with G-cases I've been running on Edison:

ERROR: module command /opt/modules/default/bin/modulecmd python rm PrgEnv-intel PrgEnv-cray PrgEnv-gnu intel cce gcc cray-parallel-netcdf cray-parallel-hdf5 pmi cray-libsci cray-mpich2 cray-mpich cray-netcdf cray-hdf5 cray-netcdf-hdf5parallel craype-sandybridge craype-ivybridge craype papi cray-petsc esmf craype failed with message:
cmdModule.c(579):ERROR:50: Cannot open file '/opt/cray/pe/modulefiles/cray-mpich/7.6.0' for 'reading'

**Comment by ndkeen** [comment]
Quick update as I'm back at keyboard: it looks like the only module that is missing is `cray-mpich/7.6.0`.  Updating this module to use `cray-mpich/7.7.0` will probably work.

This is what I'm doing:
a) Trying a few tests with _only_ changing cray-mpich from 7.6 to 7.7. This might be quickest/safest way to continue running, but it may also be a good time to do (b)
b) Testing a branch where I've updated all of the module versions -- which includes the intel compiler version. The goal here is to get master passing all of the tests with machine defaults.
c) Trying to get edison working with the original modules we were using before maintenance. There are two modules that NERSC is trying to provide us: `pe_e3sm` and `pe_archive`.  The `pe_e3sm` module was meant to give us the specific versions of modules we have been using -- even after software maintenance. The `pe_archive` module simply gives access to older modules allowing "manual" loading of older modules. However, these modules are not currently working as I would expect and I've emailed NERSC to debug.

**Comment by ndkeen** [comment]
Odd, I thought I made another update here yesterday.
The `module load pe_archive` command is now working at NERSC (after they debugged) and we can use that to gain access to the older modules, which allows us to use the same environment as before.  This branch is one way to do it:
`ndk/machinefiles/edison-pe-archive-maint10b`
I can create a PR to do this for maint-1.0 (how exactly do I do that?)

I believe you can also add the module command to env_mach_specific.xml in an existing case and resubmit

**Comment by tangq** [comment]
That's great, @ndkeen . What module commands are needed to add to env_mach_specific.xml for existing runs? I will do that to my DECK runs on hold.

@rljacob , is there a way to create a PR for maint-1.0 branch as mentioned by @ndkeen above?

**Comment by rljacob** [comment]
Yes its just like making a normal PR but you chose "maint-1.0" as the base instead of master.  The branch should start from maint-1.0 (or master prior to the v1.0 tag).

**Comment by rljacob** [comment]
See https://github.com/E3SM-Project/E3SM/pull/2459 for an example.

**Comment by ndkeen** [comment]
If you want to try editing env_mach_specific.xml, you should be able to add a single line:

```
      <command name="load">pe_archive</command>
      <command name="load">cray-mpich/7.6.0</command>
```

**Comment by rljacob** [comment]
Fixed by #2473

**Comment by rljacob** [comment]
@tangq was adding "load pe_archive" the **only** thing you had to do to keep running?  We can not get the pi-control case running on maint-1.0 without also doing whats in PR #2424
